### PR TITLE
InstructLab macOS App

### PR DESCRIPTION
**Pull request by jjasghar**
This is a proposal for creating a InstructLab Application

### CODE REVIEW

File: docs/instructlab-app.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# InstructLab App
```
Comment (leseb) [inline docs/instructlab-app.md:1]:
```suggestion
# InstructLab macOS App
```

File: docs/instructlab-app.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+## Scope
+
+This document is targeted for Mac applications, but the idea could easily be transferred to other operating systems.
```
Comment (leseb) [inline docs/instructlab-app.md:5]:
```suggestion
This document is targeted for macOS applications, but the idea could easily be transferred to other operating systems.
```

File: docs/instructlab-app.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+## Problem statement
+
+Starting InstructLab on your local laptop is hard. It requires a significant amount of `python` knowledge and terminal
+work, that is unrealistic for a non technologist to use. Having to install `git` specific versions of `python` and
+`xcode` requires a level of expertise that will create barriers of adoption to the InstructLab project.
+
+## Proposed solution
+
+[ollama][ollama] has a Mac application that is a double click installation for their server to run the commands
```
Comment (leseb) [inline docs/instructlab-app.md:15]:
```suggestion
[ollama][ollama] has a macOS application that is a double-click installation for their server to run the commands
```

File: docs/instructlab-app.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+locally. We propose creating the same "system bar" application, with the ability to run `ilab serve` in the background
+and a possible way to do `ilab chat` from said application.
+
+Having the `ilab` dog up in the system bar telling you that `ilab serve` is running, could open up the opportunity
+to have a model open to ask a `-qq` question to the local model, and even an ability to open up a "long running"
```
Comment (leseb) [inline docs/instructlab-app.md:20]:
```suggestion
to have a model open to ask a quick question to the local model, and even an ability to open up a "long-running"
```

File: docs/instructlab-app.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+conversation via a web browser or the like.
+
+## Next steps
+
+1. Create a simple MVP of starting the `ilab serve` application, with controls for the `serve` options, including
+   what model you'd like to run, i.e. Granite or Merlinite.
+2. Create an option to do a `-qq` to the via the drop down
```
Comment (leseb) [inline docs/instructlab-app.md:27]:
```suggestion
2. Create an option to ask a quick question (`-qq` option) to the via the drop-down
```

**Review by leseb — CHANGES_REQUESTED** [review]
I’m okay with the idea. Anything that can ease the adoption and improve the user experience is great. Thanks for working on this! Just a few nits.


Next time, could you push to your personal fork instead of pushing directly to the "instructlab" remote? It’s cluttering the [branches](https://github.com/instructlab/dev-docs/branches). We prefer to keep the branches clean, showing only "main" or release branches, rather than in-progress work. Thanks!

File: docs/instructlab-app.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+## Problem statement
+
+Starting InstructLab on your local laptop is hard. It requires a significant amount of `python` knowledge and terminal
+work, that is unrealistic for a non technologist to use. Having to install `git` specific versions of `python` and
+`xcode` requires a level of expertise that will create barriers of adoption to the InstructLab project.
+
+## Proposed solution
+
+[ollama][ollama] has a macOS application that is a double-click installation for their server to run the commands
+locally. We propose creating the same "system bar" application, with the ability to run `ilab serve` in the background
```
Comment (leseb) [inline docs/instructlab-app.md:16]:
```suggestion
locally. We propose creating the same "system bar" application, with the ability to run `ilab model serve` in the background
```

File: docs/instructlab-app.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+and a possible way to do `ilab chat` from said application.
```
Comment (leseb) [inline docs/instructlab-app.md:17]:
```suggestion
and a possible way to do `ilab model chat` from said application.
```

File: docs/instructlab-app.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+Having the `ilab` dog up in the system bar telling you that `ilab serve` is running, could open up the opportunity
```
Comment (leseb) [inline docs/instructlab-app.md:19]:
```suggestion
Having the `ilab` dog up in the system bar telling you that `ilab model serve` is running, could open up the opportunity
```

File: docs/instructlab-app.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+to have a model open to ask a quick question to the local model, and even an ability to open up a "long-running"
+conversation via a web browser or the like.
+
+## Next steps
+
+1. Create a simple MVP of starting the `ilab serve` application, with controls for the `serve` options, including
```
Comment (leseb) [inline docs/instructlab-app.md:25]:
```suggestion
1. Create a simple MVP of starting the `ilab model serve` application, with controls for the `serve` options, including
```

File: docs/instructlab-app.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+   what model you'd like to run, i.e. Granite or Merlinite.
+2. Create an option to ask a quick question (`-qq` option) to the via the drop-down
+3. Create a `ilab chat` type interface via a window or web browser.
```
Comment (leseb) [inline docs/instructlab-app.md:28]:
```suggestion
3. Create a `ilab model chat` type interface via a window or web browser.
```

**Review by leseb — CHANGES_REQUESTED** [review]
Just nits, missed those during my initial review, almost there thanks!

**Review by leseb — APPROVED** [review]
Thanks!

**Comment by jjasghar** [comment]
For my future self: https://medium.com/@jackhuang.wz/in-just-two-steps-you-can-turn-a-python-script-into-a-macos-application-installer-6e21bce2ee71

**Pull request merged**